cookies/operations.py

```diff
@@ -389,3 +430,58 @@ def generate_collection_coauthor_graph(collection,
         nx.set_node_attributes(graph, 'uri', node_uris)
 
     return graph
+
+@mock.patch('requests.get')
+def concept_search(text, mock_get):
+    """
+    Edit :class`.Entity` instance by linking a :class:`.ConceptEntity` URI with
+    it. This provides functionality to search for a URI based on the text
+    entered in the search field.
+
+    BlackGoat API is used to search for the URIs and the sources by querying
+    with the text entered.
+
+    Parameters
+    ----------
+    text : Search text for the URIs
+    mock_get: Mocking the functionality of BlackGoat API
+
+    Returns
+    -------
+    concepts: A list of all :class:`.GoatConcept` objects from the search result
+    """
+
+    with open('cookies/tests/data/concept_search_results.json', 'r') as f:
+            with open('cookies/tests/data/concept_search_created.json', 'r') as f2:
+                mock_get.return_value = MockSearchResponse(mock_get, f2.read(), f.read(), 200)
+
+    max_calls = 3
+    #If no query text is entered, the result from search is None.
+    if not text:
+        print 'Came here'
+        return None
+
+    #The BlackGoat API for search is used to get a list of all URIs associated
+    # with the text entered.
+    concepts = goat.Concept.search(q=text)
+
+    #RuntimeError raised when the calls are not made up to the maximum number
+    # of calls and the status code is 202.
+    if mock_get.call_count != max_calls:
+        raise RuntimeError("Should keep calling if status code 202 is received.")
+
+    #The redirect URL should be followed by the response and the number of
+    # concepts in the result should be 10.
+    args, kwargs =  mock_get.call_args
+    if args[0] != MockSearchResponse.url:
```

Comment:
``MockSearchResponse`` is presumably going to go away when you clean up this code, so what is it doing here in the middle of your function?